plugins/ccgx/fu-ccgx-dmc-device.c

```diff
@@ -616,33 +621,26 @@ fu_ccgx_dmc_device_attach(FuDevice *device, FuProgress *progress, GError **error
 		return TRUE;
 
 	if (self->update_model == DMC_UPDATE_MODEL_DOWNLOAD_TRIGGER) {
-		DmcTriggerCode trigger_code = DMC_TRIGGER_CODE_UPDATE_NOW;
-
-		if (manual_replug)
-			trigger_code = DMC_TRIGGER_CODE_UPDATE_ON_DISCONNECT;
-
-		if (!fu_ccgx_dmc_device_send_download_trigger(self, trigger_code, error)) {
-			g_prefix_error(error, "download trigger error: ");
-			return FALSE;
-		}
+		if (self->trig_code > 0) {
+			if (!fu_ccgx_dmc_device_send_download_trigger(self, self->trig_code, error)) {
+				g_prefix_error(error, "download trigger error: ");
+				return FALSE;
+			}
+			if (self->trig_code == 2) /* trigger code for update on disconnect */
```

Comment:
so I think you want to probably do something to artificially stop updates from being staged again until it's been unplugged.  I suggest adding an inhibit.

Reply:
> so I think you want to probably do something to artificially stop updates from being staged again until it's been unplugged. I suggest adding an inhibit.

Device start update firmware internally when user disconnect device from a computer.

Do I need to add an inhibit in this case?

If yes, could you let me know how to add inhibit for this case?

Reply:
Yes you should inhibit it so that a user doesn't try to run the update twice in a row before they unplug.
Here's an example: https://github.com/fwupd/fwupd/blob/da37d93eb03822e67dfb13377f14d95b27426291/plugins/dell-dock/fu-dell-dock-i2c-ec.c#L585

Reply:
Please check [53c653c](https://github.com/fwupd/fwupd/pull/4435/commits/53c653ca9db90a142b95992e167f06f819396a0f)